src/main/java/pl/edu/agh/iet/katabank/Account.java

```diff
@@ -34,7 +34,20 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-
         return Objects.hash(id);
     }
+
+    public void setBalance(BigDecimal balance) {
+        this.balance = balance;
+    }
+
+    public void withdraw(BigDecimal amount) throws IllegalArgumentException {
+        if (amount.signum()<0){
+            throw new IllegalArgumentException("The amount to withdraw is negative.");
```

Comment:
I think we should also check amount for null.
Throw exception when the amount equals 0 not only negative value. (Something to consider)
Print the amount value in the exception message.

Reply:
Code corrected.